base/math.jl

```diff
@@ -568,11 +567,18 @@ end
 Compute the hypotenuse ``\\sqrt{\\sum x_i^2}`` avoiding overflow and underflow.
 """
 hypot(x::Number...) = _hypot(promote(x...))
-# https://github.com/JuliaLang/julia/pull/30301#issuecomment-445944572
 function _hypot(x::NTuple{N,<:Number}) where N
-    simple = sum(abs2(float(y)) for y in x)
-    isinf(simple) || simple ≤ floatmin(simple) * (N-1) || return sqrt(simple)
-    return __hypot(x)
+	simple = zero(float(eltype(x)))
+	for y in x
+		yf = float(y)
+		isinf(yf) && return oftype(yf, Inf)
+		simple += abs2(yf)
+	end
+	if isinf(simple) || simple ≤ floatmin(simple) * (N-1)
+		return __hypot(x)
+	else
+		return sqrt(simple)
+	end
 end
```

Comment:
Probably faster to do something like:
```jl
simple = sum(abs2(float(y)) for y in x)
isinf(simple) || simple ≤ floatmin(simple) * (N-1) || return sqrt(simple)
isnan(simple) && return any(isinf, y) ? oftype(y, Inf) : y
return __hypot(x)
```
This way you only add one additional check in the common case where no NaN values are encountered.